### ISSUE TITLE
Touch events are not working in draw v1.4.0.  Can't draw polyline or polygon on iPhone.

### ISSUE THREAD

**Issue by jaybo** [mapbox/mapbox-gl-draw#1154]
**2.12.0**: mapbox-gl
**1.4.0**: mapbox-gl-draw

### Steps to Trigger Behavior

 1. Open Mapbox doc page:  https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-draw/
 2. Draw a polygon with the mouse.  Edit the polygon with a mouse.  All good.
 3. Try the same thing on a touch screen, either touch monitor or iPhone.  Touch events are mostly non-functional.  

### Expected Behavior
Polygon or polyline can be drawn and edited on touch devices.
### Actual Behavior
Most touch events are not processed correctly when drawing or editing polygon or polyline.

### Other
1. In my large scale mapbox-gl app I've reverted draw to v1.3.0 and everything works fine.  Using draw v1.4.0 I can't draw or edit polylines.
2. adding `touchBuffer: 25` does not improve the issue.
3. I'm guessing this is related to the `preventDefault` checkin https://github.com/mapbox/mapbox-gl-draw/pull/1146/commits/fdd235915ce640c76cb64e7b349fb59b33b16799
4. Here's another sample codepen demonstrating the issue. https://codepen.io/jaybonomad/pen/ZEjNgYj
On a desktop Windows11/Chrome touchscreen it generates a series of possibly related messages for touch events such as:
```
[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. 
Consider marking event handler as 'passive' to make the page more responsive. 
See https://www.chromestatus.com/feature/5745543795965952
```

**Comment by rhysstubbs** [comment]
I had a fork of this repo that fixed the `preventDefault` handling of touch events which has been working since I started using it. I've just reverted back to the official repo @ v1.4.0 and all of the touch events have stopped working for us too. This has been tested on IOS & Android, both Safari & Chome.

**Comment by stepankuzmin** [comment]
Thanks for reporting this issue! I've reverted the change that was causing this https://github.com/mapbox/mapbox-gl-draw/pull/1158 and released a patch version https://github.com/mapbox/mapbox-gl-draw/releases/tag/v1.4.1

**Comment by danielsippel** [comment]
Confirm, 1.4.0 broke the touch devices.

Sorry for that. Didn't realize my fix for #1019 1019 messed it up. Had some workaround lines in my code, still.
